web/packages/teleport/src/Console/ActionBar/ActionBar.tsx

```diff
@@ -43,8 +49,104 @@ export default function ActionBar(props: Props) {
   );
 }
 
+function LatencyMenuItem({ latency }: PropTypes) {
+  function colorForLatency(l: number): string {
+    if (l > 400) {
+      return 'dataVisualisation.tertiary.abbey';
+    }
+
+    if (l > 150) {
+      return 'dataVisualisation.tertiary.sunflower';
+    }
+
+    return 'dataVisualisation.tertiary.caribbean';
+  }
+
+  const totalColor = colorForLatency(latency.total);
+  const clientColor = colorForLatency(latency.client);
+  const serverColor = colorForLatency(latency.server);
+
+  return (
+    <MenuIcon Icon={Icons.Wifi} buttonIconProps={{ color: totalColor }}>
+      <Container>
+        <Flex gap={5} flexDirection="column">
+          <Text textAlign="left" typography="h3">
+            Network Connection
+          </Text>
+
+          <Flex flexDirection="row" alignItems="center">
+            <Flex mr={2} gap={1} flexDirection="column" alignItems="center">
+              <Icons.User />
+              <Text>You</Text>
+            </Flex>
+
+            <Flex mr={2} gap={1} flexDirection="column" alignItems="center">
+              <Flex mr={2} gap={1} flexDirection="row" alignItems="center">
+                <Icons.ChevronLeft size="medium" />
+                <Line></Line>
+                <Icons.ChevronRight size="medium" />
+              </Flex>
+              <Text color={clientColor}>{latency.client}ms</Text>
+            </Flex>
+
+            <Flex mr={2} gap={1} flexDirection="column" alignItems="center">
+              <TeleportGearIcon size={24}></TeleportGearIcon>
+              <Text>Teleport</Text>
+            </Flex>
+
+            <Flex mr={2} gap={1} flexDirection="column" alignItems="center">
+              <Flex mr={2} gap={1} flexDirection="row" alignItems="center">
+                <Icons.ChevronLeft size="medium" />
+                <Line></Line>
+                <Icons.ChevronRight size="medium" />
+              </Flex>
+              <Text color={serverColor}>{latency.server}ms</Text>
+            </Flex>
+
+            <Flex mr={2} gap={1} flexDirection="column" alignItems="center">
+              <Icons.Server />
+              <Text>Server</Text>
+            </Flex>
+          </Flex>
+
+          <Flex flexDirection="column" alignItems="center">
+            <Text bold fontSize={2} textAlign="center" color={totalColor}>
+              Total Latency: {latency.total}ms
+            </Text>
+          </Flex>
+        </Flex>
+      </Container>
+    </MenuIcon>
+  );
+}
+
+const Container = styled.div`
+  background: ${props => props.theme.colors.levels.surface};
+  padding: 24px;
```

Comment:
you _can_ use a props derived value here but tbh, it's not a big deal if you don't, especially if this padding isn't called out as "part of the design system". but just fyi
```suggestion
  padding: ${props => props.theme.space[4]}px;
```

You can find different values like this in the `sharedStyles.ts` file or any of the full theme files like `lightTheme.ts`